### DISCUSSION TITLE
What engine is used for http requests?

### DISCUSSION THREAD

**Discussion by daMatz** [mitmproxy/mitmproxy#5538] [Q&A] [answered]
What engine is used by mitmproxy (or mitmweb if it makes a difference) to send HTTP requests?

E.g. node is using Google V8 JS engine under the hood.

**Comment by mhils** [comment, marked as answer]
We generally work on raw Python sockets with different libraries depending on the protocol: https://docs.mitmproxy.org/dev/concepts-protocols/

**Reply by daMatz** [reply]
Thank you :)
